aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteModelTree.java

```diff
@@ -160,14 +146,34 @@ private void recurseTree(
         }
     }
 
-    private Cursor queryAll(
-            @NonNull String tableName,
-            @NonNull QueryOptions options
-    ) throws DataStoreException {
-        final ModelSchema schema = registry.getModelSchemaForModelClass(tableName);
-        final SqlCommand sqlCommand = commandFactory.queryFor(schema, options);
-        final String rawQuery = sqlCommand.sqlStatement();
-        final String[] bindings = sqlCommand.getBindingsAsArray();
-        return database.rawQuery(rawQuery, bindings);
+    private Cursor queryChildren(
+            @NonNull String childTable,
+            @NonNull String childIdField,
+            @NonNull String parentIdField,
+            @NonNull Collection<String> parentIds
+    ) {
+        // SELECT <child_id>, <parent_id> FROM <child_table> WHERE <parent_id> IN (<id_1>, <id_2>, ...)
+        String queryString = String.valueOf(SqlKeyword.SELECT) +
+                SqlKeyword.DELIMITER +
+                Wrap.inBackticks(childIdField) +
+                SqlKeyword.SEPARATOR +
+                Wrap.inBackticks(parentIdField) +
+                SqlKeyword.DELIMITER +
+                SqlKeyword.FROM +
+                SqlKeyword.DELIMITER +
+                Wrap.inBackticks(childTable) +
+                SqlKeyword.DELIMITER +
+                SqlKeyword.WHERE +
+                SqlKeyword.DELIMITER +
+                Wrap.inBackticks(parentIdField) +
+                SqlKeyword.DELIMITER +
+                SqlKeyword.IN +
+                SqlKeyword.DELIMITER +
+                Wrap.inParentheses(parentIds
+                    .stream()
```

Comment:
Let's avoid `stream()` for now. Technically as of Studio 4+, I think this gets desugared at build time. So there isn't actually any hard technical reason not to use it, anymore.

However, we don't want to mix too many different functional paradigms in the code base. Rx has a way to express most everything that the Collections APIs do -- and usually more.

FWIW, I do sort of prefer using the Stream APIs natively on the backend that support Oracle/Corretto JVM, as opposed to Rx. 🤷